lib/elixir/lib/code/formatter.ex

```diff
@@ -1362,43 +1362,57 @@ defmodule Code.Formatter do
     ["\n" | entries]
   end
 
-  defp list_interpolation_to_algebra([entry | entries], escape, state, acc, last)
-       when is_binary(entry) do
-    acc = concat(acc, escape_string(entry, escape))
-    list_interpolation_to_algebra(entries, escape, state, acc, last)
-  end
-
   defp list_interpolation_to_algebra([entry | entries], escape, state, acc, last) do
-    {{:., _, [Kernel, :to_string]}, _meta, [quoted]} = entry
-    {doc, state} = interpolation_to_algebra(quoted, state)
+    {doc, state} =
+      case entry do
+        {{:., _, [Kernel, :to_string]}, _meta, [quoted]} ->
+          interpolated_to_algebra(quoted, state)
+
+        _ ->
+          {escape_interpolation_entry(entry, escape), state}
+      end
+
     list_interpolation_to_algebra(entries, escape, state, concat(acc, doc), last)
   end
 
   defp list_interpolation_to_algebra([], _escape, state, acc, last) do
     {concat(acc, last), state}
   end
 
-  defp interpolation_to_algebra([entry | entries], escape, state, acc, last)
-       when is_binary(entry) do
-    acc = concat(acc, escape_string(entry, escape))
-    interpolation_to_algebra(entries, escape, state, acc, last)
-  end
-
   defp interpolation_to_algebra([entry | entries], escape, state, acc, last) do
-    {:"::", _, [{{:., _, [Kernel, :to_string]}, _meta, [quoted]}, {:binary, _, _}]} = entry
-    {doc, state} = interpolation_to_algebra(quoted, state)
+    {doc, state} =
+      case entry do
+        {:"::", _, [{{:., _, [Kernel, :to_string]}, _meta, [quoted]}, {:binary, _, _}]} ->
+          interpolated_to_algebra(quoted, state)
+
+        _ ->
+          {escape_interpolation_entry(entry, escape), state}
+      end
+
     interpolation_to_algebra(entries, escape, state, concat(acc, doc), last)
   end
 
   defp interpolation_to_algebra([], _escape, state, acc, last) do
     {concat(acc, last), state}
   end
 
-  defp interpolation_to_algebra(quoted, %{skip_eol: skip_eol} = state) do
+  defp interpolated_to_algebra(quoted, %{skip_eol: skip_eol} = state) do
```

Comment:
I renamed this because I found it confusing, but if anybody has another suggestion I'm happy to take it.